### PR TITLE
kie-kogito-docs-554: Guide to help DB Admins to understand Serverless Workflows persistence

### LINKED ISSUE #554: Guide to help DB Admins to understand Serverless Workflows persistence

### PR DESCRIPTION
**Pull request by wmedvede**
Closes #554 

**Description:**

Please make sure that your PR meets the following requirements:

- [X] You have read the [contributions doc](https://github.com/apache/incubator-kie-kogito-docs/blob/main/CONTRIBUTING.md)
- [X] Pull Request title is properly formatted: `Issue-XYZ Subject`
- [X] Pull Request title contains the target branch if not targeting main: `[0.9.x] Issue-XYZ Subject`
- [X] The nav.adoc file has a link to this guide in the proper category
- [X] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
+= Core Concepts
+:compat-mode!:
+// Metadata:
+:description: Running Quarkus Workflow Application using PostgresSQL
+:keywords: kogito, workflow, quarkus, serverless, quarkus-cli, persistence, postgresql
+
+The SonataFlow workflows runtime persistence is the mechanism to ensure that your workflow instances remains consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc:7]:
```suggestion
The SonataFlow workflows runtime persistence is the mechanism to ensure that your workflow instances remain consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflows runtime, and is persisted at different moments of the workflow execution.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc:9]:
```suggestion
Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflow's runtime, and is persisted at different moments of the workflow execution.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+For example, when a workflow instance reaches a state that needs to wait for an event, the engine takes a snapshot of the most relevant information, stores it in the database, and pause that instance execution.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc:11]:
```suggestion
For example, when a workflow instance reaches a state that needs to wait for an event, the engine takes a snapshot of the most relevant information, stores it in the database, and pauses that instance execution.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+:keywords: kogito, workflow, quarkus, serverless, quarkus-cli, persistence, postgresql
+:flyway_url: https://flywaydb.org/
+
+This document shows a detailed view of the database structure when you use the PostgreSQL persistence.
+
+.Prerequisites
+* You are familiar with the xref:use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc[Persistence Core concepts] guide
+* You are familiar with the xref:use-cases/advanced-developer-use-cases/persistence/persistence-with-postgresql.adoc[Persistence with PostreSQL] guide
+
+As it was introduced in the xref:use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc[Persistence Core concepts], every workflow instance requires some status information and data to execute, this information is automatically managed by the workflows runtime. And is persisted at different moments of the workflow execution in the form of a snapshot.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc:14]:
```suggestion
As it was introduced in the xref:use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc[Persistence Core concepts], every workflow instance requires some status information and data to execute, this information is automatically managed by the workflow's runtime. And is persisted at different moments of the workflow execution in the form of a snapshot.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+In the following diagram you can see the tables that support these information:
+
+image::persistence/Persistence-PostgreSQL-Advanced.png[]
+
+[NOTE]
+====
+Regular backup procedures are not performed by the workflows runtime persistence, and thus, they must be provided by the installation owner.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc:22]:
```suggestion
Regular backup procedures are not performed by the workflow's runtime persistence, and thus, they must be provided by the installation owner.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+====
+
+== `process_instances`
+This table stores the workflows snapshots. These snapshots are stored in binary format, and by default, are maintained in the database as long as the workflow instance is active. When a workflow instance finalizes it execution, the corresponding snapshot is removed from the database for optimization purposes.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc:26]:
```suggestion
This table stores the workflows snapshots. These snapshots are stored in binary format, and by default, are maintained in the database as long as the workflow instance is active. When a workflow instance finalizes its execution, the corresponding snapshot is removed from the database for optimization purposes.
```

**Review by nmirasch — APPROVED** [review]
LGTM, I've added some suggestions, feel free to take that into account!

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
+= Core Concepts
+:compat-mode!:
+// Metadata:
+:description: Running Quarkus Workflow Application using PostgresSQL
+:keywords: kogito, workflow, quarkus, serverless, quarkus-cli, persistence, postgresql
+
+The SonataFlow workflows runtime persistence is the mechanism to ensure that your workflow instances remain consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc:7]:
```suggestion
The {product_name} workflow runtime persistence is the mechanism to ensure that your workflow instances remain consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflow's runtime, and is persisted at different moments of the workflow execution.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc:9]:
```suggestion
Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflow's runtime and is persisted at different moments of the workflow execution.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+For example, when a workflow instance reaches a state that needs to wait for an event, the engine takes a snapshot of the most relevant information, stores it in the database, and pauses that instance execution.
+In this way, resources like memory are released, and can be used by other executing instances.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc:12]:
```suggestion
In this way, resources like memory are released and can be used by other executing instances.
```

**Review by ricardozanini — APPROVED** [review]
Small nitpicks :)

**Comment by ricardozanini** [comment]
@kaldesai can you please take a look? 🙏

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    correlation            json                  NOT NULL,
+    version                bigint,
+    CONSTRAINT correlation_instances_pkey PRIMARY KEY (id)
+);
+CREATE INDEX idx_correlation_instances_encoded_id ON correlation_instances (encoded_correlation_id);
+CREATE INDEX idx_correlation_instances_correlated_id ON correlation_instances (correlated_id);
+----
+
+== `flyway_schema_history`
+If the link:{flyway_url}[Flyway] database schema generation was enabled for the current workflow project, this table contains the information about the schema versions and the corresponding updates. And is completely managed by that framework.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc:62]:
```suggestion
If the link:{flyway_url}[Flyway] database schema generation was enabled for the current workflow project, this table contains information about the schema versions and the corresponding updates and that framework completely manages it.
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+* To enable the Flyway managed database schema generation you must follow this xref:use-cases/advanced-developer-use-cases/persistence/persistence-with-postgresql.adoc#create-database-schema[procedure].
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc:64]:
```suggestion
* To enable the Flyway managed database schema generation, you must follow this xref:use-cases/advanced-developer-use-cases/persistence/persistence-with-postgresql.adoc#create-database-schema[procedure].
```

File: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+image::persistence/Persistence-PostgreSQL-Advanced.png[]
+
+[NOTE]
+====
+Regular backup procedures are not performed by the workflow's runtime persistence, and thus, they must be provided by the installation owner.
+====
+
+== `process_instances`
+This table stores the workflows snapshots. These snapshots are stored in binary format, and by default, are maintained in the database as long as the workflow instance is active. When a workflow instance finalizes its execution, the corresponding snapshot is removed from the database for optimization purposes.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/postgresql-advanced-concepts.adoc:26]:
```suggestion
This table stores the workflow snapshots. These snapshots are stored in binary format, and by default, are maintained in the database as long as the workflow instance is active. When a workflow instance finalizes its execution, the corresponding snapshot is removed from the database for optimization purposes.
```

**Review by kaldesai — APPROVED** [review]
I do have very minor comments. Otherwise, your content looks very good to me :) Very well written! Thank you!

**Review by domhanak — APPROVED** [review]
Awesome guide, thanks @wmedvede

**Pull request merged**